jbake-core/src/main/java/org/jbake/app/Oven.java

```diff
@@ -186,13 +221,46 @@ private void renderContent() {
         Renderer renderer = utensils.getRenderer();
         ContentStore contentStore = utensils.getContentStore();
 
-        for (RenderingTool tool : ServiceLoader.load(RenderingTool.class)) {
+        ServiceLoader<RenderingTool> renderTools = ServiceLoader.load(RenderingTool.class);
+
+        // If this is enabled, then this already happened in Crawler.
+        // TODO: Remove the fixing from Crawler.
+        //       We should keep the pristine doc body as long as possible, or change it locally.
```

Comment:
Yes. That's a thing we should discuss. But I don't think it it's necessary to change this behavior to introduce the new Engine. It really should be integrated into the rendering phase to produce alternative paths for index files for types, that live in a totally different location than the listed document of the specific type.